src/wp-includes/nav-menu.php

```diff
@@ -716,49 +716,31 @@ function wp_get_nav_menu_items( $menu, $args = array() ) {
 		$items = array();
 	}
 
-	// Get all posts and terms at once to prime the caches.
-	if ( empty( $fetched[ $menu->term_id ] ) && ! wp_using_ext_object_cache() ) {
```

Comment:
I don't understand why the `! wp_using_ext_object_cache()` was here. I think you're right to remove it.

If anyone has a history lesson on why it was included, I'd be interested to hear it.